core/domain/suggestion_domain.py

```diff
@@ -72,3 +66,196 @@ def to_dict(self):
             'change_cmd': self.change_cmd,
             'score_category': self.score_category
         }
+
+    @classmethod
+    def from_dict(cls):
+        """Return a Suggestion object of type from a dict."""
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement from_dict.')
+
+    def validate(self):
+        """Validates the BaseSuggestion object. Each subclass must implement
+        this function
```

Comment:
Nit: add period at end.

Please also add notes on what validations exactly each subclass should implement. This will be useful when writing new suggestion subclasses later.

Reply:
Done. Also specified what needs to be validated in the subclass.